From 75c48084f1267497c5a57e476b9c448de7879a3a Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Wed, 15 Mar 2017 00:18:54 +0800
Subject: [PATCH 0888/1345] dts: a3700: add PM support for for USB2 host

commit  da6648112c6351ebb02e6c792075768a856a8d8d from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Armada 3700 usb2 host controller uses a dedicated utmi phy and ehci
  driver powers off the phy in suspend and powers on the phy in resume,
  so add the phy reference to usb node which is for usb2 host
  controller;
- Armada 3700 usb2 needs to force ehci reset after resume to insure
  data traffic continuity, so fdt property "needs-reset-on-resume" is
  set.

Change-Id: I719252726444b0a9dc5126f08a85ba8234021efd
Signed-off-by: Ken Ma <make@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37441
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-3720-community.dts     |    2 ++
 arch/arm64/boot/dts/marvell/armada-3720-db.dts     |    2 ++
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi       |    1 +
 3 files changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-3720-community.dts b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
index b75e269..7578073 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-community.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-community.dts
@@ -119,6 +119,8 @@
 
 			usb@5e000 {
 				status = "okay";
+				phys = <&utmi_usb2>;
+				phy-names = "usb";
 			};
 
 			sdhci1: sdhci@d0000 {
diff --git a/arch/arm64/boot/dts/marvell/armada-3720-db.dts b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
index c161dd3..adc9876 100644
--- a/arch/arm64/boot/dts/marvell/armada-3720-db.dts
+++ b/arch/arm64/boot/dts/marvell/armada-3720-db.dts
@@ -280,6 +280,8 @@
 /* CON27 */
 &usb2 {
 	status = "okay";
+	phys = <&utmi_usb2>;
+	phy-names = "usb";
 };
 
 
diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 5a4c729..18f74b5 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -404,6 +404,7 @@
 				compatible = "marvell,armada-3700-ehci";
 				reg = <0x5e000 0x1000>;
 				interrupts = <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
+				needs-reset-on-resume;
 				status = "disabled";
 			};
 
-- 
1.7.9.5

